src/renderer/pages/Operations/components/FlexibleMultisigShell.tsx

```diff
@@ -0,0 +1,180 @@
+import { useUnit } from 'effector-react';
+
+import { type FlexibleMultisigTransactionDS } from '@/shared/api/storage';
+import {
+  type FlexibleMultisigAccount,
+  type MultisigEvent,
+  type Signatory,
+  type SigningStatus,
+  type Wallet,
+} from '@/shared/core';
+import { useI18n } from '@/shared/i18n';
+import { Accordion, BodyText, Button, CaptionText, Header, Plate, SmallTitleText } from '@/shared/ui';
+import { Address } from '@/shared/ui-entities';
+import { contactModel } from '@/entities/contact';
+import { useMultisigEvent } from '@/entities/multisig';
+import { type ExtendedChain, useNetworkData } from '@/entities/network';
+import { SignatoryCard, signatoryUtils } from '@/entities/signatory';
+import { WalletIcon, permissionUtils, walletModel } from '@/entities/wallet';
+import { getSignatoryName, getSignatoryStatus } from '../common/utils';
+
+import { Status } from './Status';
+import ApproveTxModal from './modals/ApproveTx';
+import RejectTxModal from './modals/RejectTx';
+
+type Props = {
+  tx: FlexibleMultisigTransactionDS;
+  account: FlexibleMultisigAccount;
+};
+
+export const FlexibleMultisigShell = ({ tx, account }: Props) => {
+  const { t } = useI18n();
+  const { connection, extendedChain } = useNetworkData(tx.chainId);
```

Comment:
Any chance to avoid `useNetworkData`?

Reply:
I'm trying to fix it right now, it leads to some refactoring on Operations page, but we wanted it anyway